src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -49,23 +49,26 @@ public static void putDocument(String indexName, Document document) {
             try {
                 PutResponse result = index.put(document);
 
-                if (Config.PERSISTENCE_CHECK_DURATION > 0) {
-                    int elapsedTime = 0;
-                    boolean isSuccessful = result.getResults().get(0).getCode() == StatusCode.OK;
-                    while (!isSuccessful && elapsedTime < Config.PERSISTENCE_CHECK_DURATION) {
-                        ThreadHelper.waitBriefly();
-                        // retry putting the document
-                        result = index.put(document);
-                        isSuccessful = result.getResults().get(0).getCode() == StatusCode.OK;
-                        // check before incrementing to avoid boundary case problem
-                        if (!isSuccessful) {
-                            elapsedTime += ThreadHelper.WAIT_DURATION;
-                        }
-                    }
-                    if (elapsedTime >= Config.PERSISTENCE_CHECK_DURATION) {
-                        log.info(String.format(ERROR_EXCEED_DURATION, document, indexName));
+                if (Config.PERSISTENCE_CHECK_DURATION == 0) {
+                    continue;
+                }
```

Comment:
Even if the persistence check is not needed, we still need the exponential backoff loop to handle transient backend errors ([reference](https://cloud.google.com/appengine/docs/standard/java/search/#Java_Putting_documents_in_an_index)).